### ISSUE TITLE
inline json support for the enrich mediator is not working in tool

### ISSUE THREAD

**Issue by sdkottegoda** [wso2/product-ei#3318]
**Description:**
I have created an api with the following configuration:
```<enrich>
            <source type="inline" clone="false">[{}]</source>
            <target xpath="json-eval($.input.quote.product.assets[?(@.code==$.code)].instances)"/>
    </enrich>
```
The source view of the tool doesn't allow the user to save the configuration and failes with the following error:
```
org.apache.axiom.om.OMException: com.ctc.wstx.exc.WstxUnexpectedCharException: Unexpected character 'o' (code 111) in prolog; expected '<'
 at [row,col {unknown-source}]: [1,1]
	at org.apache.axiom.om.impl.builder.StAXOMBuilder.next(StAXOMBuilder.java:296)
	at org.apache.axiom.om.impl.llom.OMDocumentImpl.getOMDocumentElement(OMDocumentImpl.java:109)
	at org.apache.axiom.om.impl.builder.StAXOMBuilder.getDocumentElement(StAXOMBuilder.java:570)
	at org.apache.axiom.om.impl.builder.StAXOMBuilder.getDocumentElement(StAXOMBuilder.java:566)
	at org.apache.axiom.om.util.AXIOMUtil.stringToOM(AXIOMUtil.java:54)
	at org.apache.axiom.om.util.AXIOMUtil.stringToOM(AXIOMUtil.java:39)
	at org.wso2.developerstudio.eclipse.gmf.esb.internal.persistence.EnrichMediatorTransformer.createEnrichMediator(EnrichMediatorTransformer.java:124)
	at org.wso2.developerstudio.eclipse.gmf.esb.internal.persistence.EnrichMediatorTransformer.transform(EnrichMediatorTransformer.java:55)
	at org.wso2.developerstudio.eclipse.gmf.esb.internal.persistence.AbstractEsbNodeTransformer.doTransform(AbstractEsbNodeTransformer.java:79)
	at org.wso2.developerstudio.eclipse.gmf.esb.internal.persistence.PayloadFactoryMediatorTransformer.transform(PayloadFactoryMediatorTransformer.java:54)
	at org.wso2.developerstudio.eclipse.gmf.esb.internal.persistence.AbstractEsbNodeTransformer.doTransform(AbstractEsbNodeTransformer.java:79)
	at org.wso2.developerstudio.eclipse.gmf.esb.internal.persistence.CallMediatorTransformer.transform(CallMediatorTransformer.java:61)
	at org.wso2.developerstudio.eclipse.gmf.esb.internal.persistence.AbstractEsbNodeTransformer.doTransform(AbstractEsbNodeTransformer.java:79)
	at org.wso2.developerstudio.eclipse.gmf.esb.internal.persistence.PropertyMediatorTransformer.transform(PropertyMediatorTransformer.java:64)
	at org.wso2.developerstudio.eclipse.gmf.esb.internal.persistence.AbstractEsbNodeTransformer.doTransform(AbstractEsbNodeTransformer.java:79)
	at org.wso2.developerstudio.eclipse.gmf.esb.internal.persistence.APIResourceTransformer.transform(APIResourceTransformer.java:179)
	at org.wso2.developerstudio.eclipse.gmf.esb.internal.persistence.APITransformer.transform(APITransformer.java:71)
	at org.wso2.developerstudio.eclipse.gmf.esb.internal.persistence.DefaultEsbModelExporter.transformAPI(DefaultEsbModelExporter.java:326)
	at org.wso2.developerstudio.eclipse.gmf.esb.internal.persistence.DefaultEsbModelExporter.designToSource(DefaultEsbModelExporter.java:503)
	at org.wso2.developerstudio.eclipse.gmf.esb.diagram.part.EsbMultiPageEditor.updateAssociatedXMLFile(EsbMultiPageEditor.java:906)
	at org.wso2.developerstudio.eclipse.gmf.esb.diagram.part.EsbMultiPageEditor.access$16(EsbMultiPageEditor.java:895)
	at org.wso2.developerstudio.eclipse.gmf.esb.diagram.part.EsbMultiPageEditor$8.run(EsbMultiPageEditor.java:1085)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:37)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:182)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4577)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4186)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1150)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1039)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:153)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:680)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:594)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:151)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:653)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:590)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1499)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1472)
Caused by: com.ctc.wstx.exc.WstxUnexpectedCharException: Unexpected character 'o' (code 111) in prolog; expected '<'
 at [row,col {unknown-source}]: [1,1]
	at com.ctc.wstx.sr.StreamScanner.throwUnexpectedChar(StreamScanner.java:639)
	at com.ctc.wstx.sr.BasicStreamReader.nextFromProlog(BasicStreamReader.java:2052)
	at com.ctc.wstx.sr.BasicStreamReader.next(BasicStreamReader.java:1134)
	at org.apache.axiom.om.impl.builder.StAXOMBuilder.parserNext(StAXOMBuilder.java:681)
	at org.apache.axiom.om.impl.builder.StAXOMBuilder.next(StAXOMBuilder.java:214)
	... 47 more
```

**Comment by Posejdon23** [comment]
Is this fix only available through the commercial WUM updates or will be added to the public release of 6.5.0/6.5.1/6.6 ?

**Comment by sdkottegoda** [comment]
Since the fix has been added in EI 6.5.0-M3, it will be available in all public releases from EI 6.5.0 onwards.
